### ISSUE TITLE
Split doesn't respect type indices

### ISSUE THREAD

**Issue by googleson78** [haskell/haskell-language-server#1710]
Version:
```
0227133dc5651cdc2da367923f5a27592465de46
```
(current master)

```haskell
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE GADTs #-}

data A b where
  ATrue :: A True
  AFalse :: A False

f :: A True -> ()
f x = _
```
Doing a case split here generates both cases, even though one of them is unreachable.

I would expect only `ATrue` to be generated.

**Comment by isovector** [comment]
I can't reproduce this. The result on my machine:

```haskell
f :: A True -> ()
f ATrue = _
```

**Comment by isovector** [comment]
Ah, this was fixed in #1733: 

https://github.com/haskell/haskell-language-server/commit/ef902894ef829bd7158a4263423ca1e1021de28c#diff-343e22cc186f5a0f1a02eaabd5ad1126135512fac7cb67f9e535511ef2b2bb7eR59
